Expired/Views/ProductImage.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ProductImage.swift
+//  Expired
+//
+//  Created by satgi on 2023-03-22.
+//
+
+import SwiftUI
+
+enum ImageSize {
+    case small
+    case medium
+    case large
+    case ultraLarge
+}
+
+struct ProductImage: View {
+    var image: UIImage? = nil
+    var data: Data? = nil
+    var size: ImageSize = .medium
+
+    var width: CGFloat {
+        var width = 50.0
+        switch size {
+            case .small:
+                width = 50.0
+            case .medium:
+                width = 100.0
+            case .large:
+                width = 150.0
+            case .ultraLarge:
+                width = 250.0
+        }
+
+        return width
+    }
+    
+    var displayImage: UIImage? {
+        if let image = image {
+            return image
+        } else if let data = data {
+            return UIImage(data: data)!
+        }
+
+        return nil
+    }
+
+    var body: some View {
+        if displayImage != nil {
+            Image(uiImage: displayImage!)
+                .resizable()
+                .aspectRatio(contentMode: .fill)
+                .frame(width: width, height: width, alignment: .center)
```

Comment:
```suggestion
                .frame(width: width, alignment: .center)
```

We can now remove the `height` parameter since we are using the width variable on the same one.  Having just the width parameter it still sets the width of the image to the width variable and keeps the aspect ratio of the image.

Reply:
We want `height` in this case as we need the image to be a square all the time.
The reason is that we can't control the ratio of width/height since the photo is uploaded by the user. Say the width/height ratio is 1:200, the photo will look very long as the height will be `width * 200` in this case, meaning the image will take too much space vertically. See attached pictures for example.

To make it looks good or consistent, it's better to make the size fixed.

![1](https://user-images.githubusercontent.com/122644881/227738745-ef2a5c1c-6f2c-4491-88c7-6646f7490bd0.jpeg)

![2](https://user-images.githubusercontent.com/122644881/227738754-addf778c-45b6-41c8-b983-0c20f0ae9ac8.jpeg)

Reply:
Makes sense. And we need consistency on each picture we take. Thanks for the explanation.